File: searchProject_java/src/test/java/StemmerTest.java
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class StemmerTest {
    public Stemmer stemmer;
    @BeforeEach
    public void stemmerInit(){
        stemmer = new Stemmer();
    }
    @Test
    public void test(){
        Assertions.assertEquals("optim", stemmer.stemWord("optimization"));
    }
    @Test
    public void endingWithEdTest(){
        Assertions.assertEquals("work", stemmer.stemWord("worked"));
    }
    @Test
    public void endingWithConsTest(){
        Assertions.assertEquals("debug", stemmer.stemWord("debugging"));
    }
    @Test
    public void pluralWithSTest(){
        Assertions.assertEquals("car", stemmer.stemWord("cars"));
    }
    @Test
    public void endingWithAbleTest(){
        Assertions.assertEquals("compat", stemmer.stemWord("compatible"));
    }
    @Test
    public void endingWithIzationTest(){
        Assertions.assertEquals("final", stemmer.stemWord("finalization"));
    }
    @Test
    public void endingWithAtionAndAtorTest(){
        Assertions.assertEquals("alloc", stemmer.stemWord("allocation"));
        Assertions.assertEquals("motiv", stemmer.stemWord("motivator"));
    }
    @Test
    public void endingWithFulnessTest(){
        Assertions.assertEquals("aw", stemmer.stemWord("awfulness"));
    }
    @Test
    public void endingWithETest(){
        Assertions.assertEquals("ablaz", stemmer.stemWord("ablaze"));
    }

}

File: searchProject_csharp/SampleLibrary/SampleLibrary.Test/StemmerTest.cs
using Xunit;

namespace SampleLibrary.Test;

public class StemmerTest
{
    private Stemmer _stemmer = new Stemmer();
    [Fact]
    public void test(){
        Assert.Equal("optim", _stemmer.StemWord("optimization"));
    }
    [Fact]
    public void endingWithEdTest(){
        Assert.Equal("work", _stemmer.StemWord("worked"));
    }
    [Fact]
    public void endingWithConsTest(){
        Assert.Equal("debug", _stemmer.StemWord("debugging"));
    }
    [Fact]
    public void pluralWithSTest(){
        Assert.Equal("car", _stemmer.StemWord("cars"));
    }
    [Fact]
    public void endingWithAbleTest(){
        Assert.Equal("compat", _stemmer.StemWord("compatible"));
    }
    [Fact]
    public void endingWithIzationTest(){
        Assert.Equal("final", _stemmer.StemWord("finalization"));
    }
    [Fact]
    public void endingWithAtionTest(){
        Assert.Equal("alloc", _stemmer.StemWord("allocation"));
    }[Fact]
    public void endingWithAtorTest(){
        Assert.Equal("motiv", _stemmer.StemWord("motivator"));
    }
    [Fact]
    public void endingWithFulnessTest(){
        Assert.Equal("aw", _stemmer.StemWord("awfulness"));
    }
    [Fact]
    public void endingWithETest(){
        Assert.Equal("ablaz", _stemmer.StemWord("ablaze"));
    }
}